kubernetes/kubernetes.go

```diff
@@ -492,13 +545,55 @@ func (ka KubeAgentConfig) sendMetricsToCustomS3(metricSample *os.File) {
 	}
 }
 
-// generateSampleKey creates a key (location) for s3 to upload the sample to. Example of s3 location format
+func (ka KubeAgentConfig) sendMetricsToCustomBlob(metricSample *os.File) {
+	cred, err := azidentity.NewClientSecretCredential(ka.CustomAzureTenantID, ka.CustomAzureClientID,
+		ka.CustomAzureClientSecret, nil)
+	if err != nil {
+		log.Fatalf("Could not establish Azure credentials, "+
+			"ensure Azure environment variables are set correctly: %s", err)
+	}
+	azureClient, err := azblob.NewClient(ka.CustomAzureBlobURL, cred, nil)
+	if err != nil {
+		log.Fatalf("Could not establish Azure Client, "+
+			"ensure Azure environment variables are set correctly: %s", err)
+	}
+	ka.uploadBlob(azureClient, metricSample)
+}
+
+func (ka KubeAgentConfig) uploadBlob(client *azblob.Client, metricSample *os.File) {
+	file, err := os.Open(metricSample.Name())
+	if err != nil {
+		log.Fatalf("Unable to open metric sample file %v", err)
+	}
+	defer file.Close()
+	key := generateSampleKey(ka.clusterUID, azure)
+	_, err = client.UploadFile(context.Background(), ka.CustomAzureUploadBlobContainerName, key, file, nil)
+	if err != nil {
+		log.Fatalf("Failed to put Object to custom Azure blob with error: %s", err)
+	} else {
+		sn := strings.Split(metricSample.Name(), "/")
```

Comment:
Why bother trimming off the suffix? Seems more reasonable to keep the full path since that's what would be in the blob storage

Reply:
This is to match the same formatting as the default mode: https://github.com/cloudability/metrics-agent/blob/220145f405967b0641d31553625152538d7c3600/kubernetes/kubernetes.go#L617